### PR TITLE
Docs: top-level refresh and restructure

### PR DESCRIPTION
**Pull request by lornajane**
Revisiting our CLI docs, with more of an end-user perspective and aiming for simpler/clearer documentation.

Main sections of work planned

- [x] Manage OpenAPI files (split/join/bundle), useful for API producers
- [x] Ensure good API governance (lint, decorate, preprocess), useful for API producers, maybe also for consumers, and wider compliance
- [x] Build beautiful docs (build-docs, preview-docs), for API producers, but also consumers if no good docs are provided

We also need a better way to preview docs locally and check links, since I'm unsure about the formatting and very sure I'm breaking things :) Probably this is needed for us to be able to review this properly, but I'll keep working on content (hopefully the commit history helps explain what I'm thinking)

Not included this time (because this PR is already too big):
 - Config docs
 - Custom plugin docs
 - Decorators
 - Migration guides from previous CLI tools/versions

### CODE REVIEW

File: docs/guides/update-cli.md
Comment (adamaltman) [inline docs/guides/update-cli.md:1]: Add the front matter with `redirectFrom` with the old url `/docs/cli/update-cli/`

**Review by adamaltman — COMMENTED** [review]
Nice 🎉 

I like it.

File: docs/guides/configure-rules.md
Comment (lornajane) [reply docs/guides/configure-rules.md:59]: This is interesting to me but I think it's too much to add in this type of article. It's a guide, so it should present one way to complete a task, without options.

File: docs/guides/update-cli.md
Comment (lornajane) [reply docs/guides/update-cli.md:1]: Got it, thanks!

File: docs/rules/paths-kebab-case.md
Comment (adamaltman) [inline docs/rules/paths-kebab-case.md:80]: Should these change everywhere?
```suggestion
- [configurable rules](./configurable-rules.md)
```

**Review by adamaltman — COMMENTED** [review]
I wanted to link to a particular document, and realized this wasn't merged yet. I reviewed it thoroughly. I left some suggestions and questions. This is great 🎉 

I think we still need to finalize the naming for configurable rules and use it consistently. One of the reasons we renamed it to custom rules is that when we changed the syntax it became composed for three parts: Subject, Where (optional), and Assertion (and it was strange that the parent object was also called an assertion -- that is the original motivator for renaming the parent object from "Assertions" to "Custom rules"). However, I also think we may want to not use the name "Custom plugins" and call them "Plugins" instead.

File: docs/rules.md
Comment (lornajane) [reply docs/rules.md:7]: Ah, yes, my English is rather mid-atlantic in places because I'm constantly switching dialects. Better tooling will pick this up for us instantly though, and save everyone time.

**Comment by lornajane** [comment]
I'd also really like to make the bullets at the top of the index page into cards or something, if anyone can add that?

**Comment by adamaltman** [comment]
> "Bullets --> Cards"

Let's do this as an update _after_ this is merged.

File: docs/api-standards.md
Comment (lornajane) [reply docs/api-standards.md:7]: artistic license ;)

File: docs/guides/configure-rules.md
Comment (lornajane) [reply docs/guides/configure-rules.md]: Good tip!

File: docs/rules/paths-kebab-case.md
Comment (lornajane) [reply docs/rules/paths-kebab-case.md:80]: Yes, but I can probably `sed` it, I did the links this way (which is how I didn't spot the words!)

**Review by lornajane — COMMENTED** [review]
Always slightly confused at being able to comment on my own PR :)

File: docs/rules/configurable-rules.md
Comment (lornajane) [reply docs/rules/configurable-rules.md:20]: I'm starting to think that the main confusion comes from the `assert/give-the-rule-a-name` syntax. Because THAT is a rule, and it contains assertions and other criteria. I don't want to block this docs PR for a syntax change though, so I'll just try to fix the wording on what we have, and look at whether we could use `rule/my-better-rule` naming instead without too much disruption.

File: docs/rules/configurable-rules.md
Comment (lornajane) [reply docs/rules/configurable-rules.md:1]: Got it!

File: docs/rules/configurable-rules.md
Comment (lornajane) [reply docs/rules/configurable-rules.md]: I've moved this to #1061 so that we can track it there, and not hold up the docs.

File: docs/rules/paths-kebab-case.md
Comment (lornajane) [reply docs/rules/paths-kebab-case.md]: These are fixed now.

**Comment by lornajane** [comment]
In an attempt to get this PR mergeable (it's already HUGE), I've moved a couple of discussions to separate issues
- #1060 for a better landing page layout with cards
- #1061 to consider naming of assert things

File: docs/rules/builtin-rules.md
Comment (lornajane) [reply docs/rules/builtin-rules.md:1]: We could, but I was leaning towards using hyphens to replace spaces, and not replacing in-word punctuation like hyphens or apostrophes or whatever. If it's important I can change it and update all the links.

**Comment by lornajane** [comment]
@adamaltman along our naming discussion, I opened https://github.com/Redocly/redocly-cli/pull/1062 to propose using `rule/i-made-a-rule` as the configurable rules key in the config file. You may have opinions.

File: docs/guides/configure-rules.md
Comment (lornajane) [reply docs/guides/configure-rules.md:155]: I think just "plugins" is actually a really good idea, since we don't publish our own (by definition, ours are inside the product!). Also the rules aren't that custom, other than the custom function ones, so I think configurable rules works as well as any of the other options.

(Plugins docs need some work but in a separate PR!)

File: docs/commands/index.md
Comment (adamaltman) [inline docs/commands/index.md:52]:
```suggestion
The Redocly configuration file must sit in your root directory.
```
Seems like a remnant related to using a starter template repo.

File: docs/quickstart.md
Comment (adamaltman) [inline docs/quickstart.md:22]: question: do we want to enforce no past/future tense?
```suggestion
The output takes a few moments to build, and then lets you know the host and port where the preview server is running - usually `http://localhost:8080`.
```

File: docs/rules.md
Comment (adamaltman) [inline docs/rules.md:11]:
```suggestion
* **Configurable rules** allow powerful describing of API standards without needing to write code. Create a configurable rule, choose which parts of the OpenAPI description it applies to, and what the criteria for success are. The linting tool does the rest. With plenty of examples, the [configurable rules](./rules/configurable-rules.md) helps you to describe your API standards easily and well.
```
Same question about the future tense here.

File: docs/rules/builtin-rules.md
Comment (adamaltman) [reply docs/rules/builtin-rules.md:1]: My concern is it if has any impact (even tiny) on SEO.

File: docs/rules/recommended.md
Comment (adamaltman) [inline docs/rules/recommended.md:1]:
```suggestion
# Recommended ruleset
```

**Review by adamaltman — APPROVED** [review]
I think this PR is good to merge once https://github.com/Redocly/redocly-cli/pull/1075 merges to change the built-in rule filename and links.

One other thing that needs to happen shortly thereafter is the `sidebars.yaml` file needs to be updated in the marketing site with these corresponding file changes too.

**Pull request merged**
